### PR TITLE
feat(file): Added download attributes. - OEL-2772

### PR DESCRIPTION
**Pull request by tibi2303**

### CODE REVIEW

**Review by AaronGilMartinez — CHANGES_REQUESTED** [review]

File: src/compositions/bcl-file/file.html.twig [hunk 1]
```diff
@@ -138,16 +138,15 @@
...
+          icon: _download.icon|default({}|merge({
             name: "download",
             size: "fluid",
             path: _icon_path,
-          },
-          attributes: create_attribute()
-                      .setAttribute('download', true)
-                      .setAttribute('target', '_blank')
+          })),
+          attributes: _download.attributes|default(create_attribute().merge(download_attributes))
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/compositions/bcl-file/file.html.twig]: done, changed the approach as we discussed.

**Review by AaronGilMartinez — APPROVED** [review]

File: src/compositions/bcl-file/file.html.twig [hunk 2]
```diff
@@ -138,17 +138,18 @@
...
+          icon: _download.icon|default({}|merge({
             name: "download",
             size: "fluid",
             path: _icon_path,
-          },
-          attributes: create_attribute()
-                      .setAttribute('download', true)
-                      .setAttribute('target', '_blank')
-                      .addClass(['standalone', 'align-self-center', 'd-inline-block', 'mt-1', 'mt-md-0', 'flex-shrink-0'])
+          })),
```
Comment (drishu) [inline src/compositions/bcl-file/file.html.twig:151]: This is too confusing for me, we need to have defaults and merge over what is coming as parameter, so
```
icon: {
              name: "download",
              size: "fluid",
              path: _icon_path,
            }|merge(_download.icon),
```

File: src/compositions/bcl-file/file.test.js [hunk 3]
```diff
@@ -87,9 +87,57 @@ describe("OE - File Card", () => {
...
 
+  test(`renders correctly with different download icon`, () => {
+    expect.assertions(1);
+    return expect(
+      render({
+        ...demoCardData,
+        download: {
+          label: "Download",
+          path: "/example.html",
+          icon: {
```
Comment (drishu) [inline src/compositions/bcl-file/file.test.js:98]: this is the only scenatio with icon, what if I send an empty icon, or one with just the "path"

**Review by drishu — CHANGES_REQUESTED** [review]

File: src/compositions/bcl-file/file.html.twig [hunk 4]
```diff
@@ -138,17 +138,18 @@
           </small>
           {% endif %}
         </div>
+        {% set download_attributes = create_attribute()
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/compositions/bcl-file/file.html.twig:141]: done.

File: src/compositions/bcl-file/file.html.twig (hunk 2 above)
Comment (tibi2303) [reply src/compositions/bcl-file/file.html.twig]: indeed, thanks for the improvement, added.

File: src/compositions/bcl-file/file.test.js (hunk 3 above)
Comment (tibi2303) [reply src/compositions/bcl-file/file.test.js:98]: added multiple tests, thanks.

**Review by drishu — APPROVED** [review]

**Review by AaronGilMartinez — APPROVED** [review]

**Pull request merged**